tests/integration/test_rest_catalog.py

```diff
@@ -16,34 +16,788 @@
+    "last-partition-id": 1000,
+    "default-sort-order-id": 1,
+    "sort-orders": [
+        {
+            "order-id": 1,
+            "fields": [
+                {"source-id": 2, "transform": "identity", "direction": SortDirection.ASC, "null-order": NullOrder.NULLS_FIRST}
+            ],
+        }
+    ],
+    "properties": {
+        "owner": "fokko",
+        "write.parquet.compression-codec": "zstd",
+    },
+    "refs": {},
+    "snapshots": [],
+    "snapshot-log": [],
+    "metadata-log": [],
+}
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_v2
+
 
-TEST_NAMESPACE_IDENTIFIER = "TEST NS"
+@pytest.fixture
+def table_metadata_no_snapshot_partitioned_v2() -> Dict[str, Any]:
+    return EXAMPLE_table_metadata_no_snapshot_partitioned_v2
+
+
+@pytest.fixture
+def rest_integration_example_metadata_partitioned_v2(table_metadata_no_snapshot_partitioned_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_partitioned_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def example_table_metadata_with_no_location(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
+
+
+@pytest.fixture
+def table_metadata_no_snapshot_rest(table_metadata_no_snapshot_v2: Dict[str, Any]) -> Dict[str, Any]:
+    return {
+        "metadata-location": f"s3://warehouse/{TEST_TABLE_IDENTIFIER[0]}/{TEST_TABLE_IDENTIFIER[1]}",
+        "metadata": table_metadata_no_snapshot_v2,
+        "config": {
+            "client.factory": "io.tabular.iceberg.catalog.TabularAwsClientFactory",
+            "region": "us-west-2",
+        },
+    }
 
 
 @pytest.mark.integration
-@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog")])
-def test_namespace_exists(catalog: RestCatalog) -> None:
-    if not catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+@pytest.fixture(scope="function")
+@pytest.mark.parametrize("catalog", [pytest.lazy_fixture("session_catalog"), pytest.lazy_fixture("test_clean_up")])
+def test_clean_up(catalog: RestCatalog) -> None:
+    for namespaces_tuple in catalog.list_namespaces():
+        namespace_name = namespaces_tuple[0]
+        if TEST_NAMESPACE_IDENTIFIER[0] in namespace_name:
+            for identifier in catalog.list_tables(namespace_name):
+                catalog.purge_table(identifier)
+            if catalog.namespace_exists(TEST_NAMESPACE_IDENTIFIER):
+                catalog.drop_namespace(TEST_NAMESPACE_IDENTIFIER)
```

Comment:
since we only have 2 tables and 1 namespace:
```
TEST_NAMESPACE_IDENTIFIER = ("rest_integration_ns",)
TEST_TABLE_IDENTIFIER = ("rest_integration_ns", "rest_integration_tbl")
TEST_TABLE_IDENTIFIER_RENAME = ("rest_integration_ns", "renamed_rest_integration_tbl")
```

we can drop all tables in the `TEST_NAMESPACE_IDENTIFIER` namespace, and then remove the remove